lib/plugin-api/hydrogen-kernel.js

```diff
@@ -37,7 +42,10 @@ export default class HydrogenKernel {
    */
   getConnectionFile() {
     this._assertNotDestroyed();
-    const { connectionFile } = this._kernel;
+
+    const connectionFile = this._kernel.connectionFile
+      ? this._kernel.connectionFile
+      : null;
     if (!connectionFile) {
```

Comment:
Only the ZMQKernel has a connectionFile prop, so importing ZMQKernel and handling this way made flow happy.